src/main/services/token/TransferAllTokensService.kt

```diff
@@ -0,0 +1,64 @@
+package main.services.token
+
+import kotlinserverless.framework.services.SOAResult
+import kotlinserverless.framework.services.SOAResultType
+import main.daos.*
+import main.daos.Transaction
+import main.helpers.TransferTokenHelper
+import main.services.reward.DistributeRewardService
+
+/**
+ * Transfer all tokens held by one address to another
+ */
+object TransferAllTokensService {
+    fun execute(
+            caller: UserAccount,
+            from: String,
+            to: String,
+            notes: String?) : SOAResult<TransactionList> {
+        val address = caller.cryptoKeyPair.publicKey
+
+        // verify that the caller is the from address
+        if(address != from)
+            return SOAResult(
+                    SOAResultType.FAILURE,
+                    "Access denied. Caller and from address must match.",
+                    null)
+
+        // Generate map of all tokenIds with a balance
+        val transactionsResult = TransferTokenHelper.getTransferHistory(address, null)
+        if(transactionsResult.result != SOAResultType.SUCCESS)
+            return SOAResult(SOAResultType.FAILURE, transactionsResult.message)
+        val mapOfTransfers = TransferTokenHelper.getMapOfTransfersByCurrency(transactionsResult.data!!)
+        val mapOfBalances = TransferTokenHelper.getMapOfBalancesByCurrency(address, mapOfTransfers)
+
+
+        // Transfer all tokens to the new address
+        var resultingTransactions = mutableListOf<Transaction>()
+        var result : SOAResult<TransactionList> =
+                SOAResult(SOAResultType.SUCCESS, "All tokens transferred successfully.", null)
+
+        if(mapOfBalances.size == 0)
+            result.message = "Address has no associated balances so no tokens were transferred."
+
+        mapOfBalances.forEach { tokenId, balance ->
+            if(balance <= 0.0 || result.result != SOAResultType.SUCCESS)
```

Comment:
if the balance is less than 0 should we change result to failure and add message?